internal/view/owner_extender.go

```diff
@@ -0,0 +1,233 @@
+package view
+
+import (
+	"errors"
+	"fmt"
+	"github.com/derailed/k9s/internal/client"
+	"github.com/derailed/k9s/internal/ui"
+	"github.com/derailed/tcell/v2"
+	"github.com/derailed/tview"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const selectOwnerDialogKey = "owner"
+
+// OwnerExtender adds log actions to a given viewer.
+type OwnerExtender struct {
+	ResourceViewer
+}
+
+// NewOwnerExtender returns a new extender.
+func NewOwnerExtender(v ResourceViewer) ResourceViewer {
+	o := OwnerExtender{
+		ResourceViewer: v,
+	}
+	o.AddBindKeysFn(o.bindKeys)
+
+	return &o
+}
+
+// BindKeys injects new menu actions.
+func (o *OwnerExtender) bindKeys(aa ui.KeyActions) {
+	aa.Add(ui.KeyActions{
+		ui.KeyO: ui.NewKeyAction("Show Owner", o.ownerCmd(false), true),
+	})
+}
+
+func (o *OwnerExtender) ownerCmd(prev bool) func(evt *tcell.EventKey) *tcell.EventKey {
+	return func(evt *tcell.EventKey) *tcell.EventKey {
+		path := o.GetTable().GetSelectedItem()
+		if path == "" {
+			return nil
+		}
+		if !isResourcePath(path) {
+			path = o.GetTable().Path
+		}
+		o.showOwner(o.GetTable().GVR(), path, prev, evt)
```

Comment:
It might be better to just have showOwner return an error and just flash the error from here.